<?xml version="1.0" ?>
<launch>
  <arg name="map_name" default="$(find mcc)/maps/simple.yaml" />

  <include file="$(find mcc)/launch/rviz_nav.launch" />

  <node pkg="map_server" type="map_server" name="map_server" output="screen" args="$(arg map_name)" />

  <node name="rviz" pkg="rviz" type="rviz" args="-d $(find mcc)/rviz/default.rviz" />

  <node pkg="mcc" type="mcc_model_node" name="mcc_model_node" output="screen">
    <param name="update_rate" value="10" />

    <param name="grid_width" value="3" />
    <param name="grid_height" value="3" />

    <param name="base_probability_of_action_success" value="0.9" />
    <param name="penalty_for_freespace" value="0.05" />

    <param name="sub_occupancy_grid" value="/map" />

    <param name="sub_map_pose_estimate" value="/initialpose" />
    <param name="sub_map_nav_goal" value="/move_base_simple/goal" />

    <param name="get_action" value="~get_action" />
    <param name="get_fsc_state" value="~get_fsc_state" />
    <param name="update_fsc_state" value="~update_fsc_state" />

    <param name="visualize_fsc_state" value="True" />
    <param name="visualization_marker" value="~visualization_marker" />
    <param name="visualization_marker_array" value="~visualization_marker_array" />
  </node>

  <node pkg="mcc" type="mcc_kobuki_node" name="mcc_kobuki_node" output="screen">
    <!-- Note "remap" for commands 'pushes' data from something to something else. Conversely,
         information 'flows' from something to something else. Hence the remapping below. -->
    <remap from="/base_pose_ground_truth" to="/odom" /> <!-- nav_msgs -->
    <remap from="/cmd_vel" to="/mobile_base/commands/velocity" /> <!-- kobuki_msgs -->
    <remap from="/cmd_reset_odom" to="/mobile_base/commands/reset_odometry" /> <!-- kobuki_msgs -->
    <remap from="/mobile_base/events/bumper" to="/evt_bump" /> <!-- kobuki_msgs -->

    <param name="mcc_model_namespace" value="/mcc_model_node" />

    <param name="at_position_goal_threshold" value="0.05" />
    <param name="at_theta_goal_threshold" value="0.05" />
    <param name="recovery_distance_threshold" value="0.125" />

    <param name="pid_integrator_bounds" value="0.05" />

    <param name="pid_theta_Kp" value="2.0" />
    <param name="pid_theta_Ki" value="0.5" />
    <param name="pid_theta_Kd" value="0.5" />

    <param name="desired_velocity" value="0.2" />

    <param name="sub_kobuki_odom" value="/odom" />
    <param name="sub_kobuki_bump" value="/evt_bump" />
    <param name="pub_kobuki_vel" value="/cmd_vel" />
    <param name="pub_kobuki_reset_odom" value="/cmd_reset_odom" />

    <param name="sub_map_pose_estimate" value="/initialpose" />
    <param name="sub_map_nav_goal" value="/move_base_simple/goal" />
  </node>
</launch>

